File: algorithms/interview-preparation-kit-1/repeated-string/ruby/response/spec/repeated_string.spec.rb
require_relative "../src/repeated_string"

describe RepeatedString do
  describe ".repeat" do
    context "('a', 10)" do
      it "10" do
        expect(RepeatedString.repeat('a', 10)).to eql(10)
      end
    end
    context "('b', 10)" do
      it "0" do
        expect(RepeatedString.repeat('b', 10)).to eql(0)
      end
    end
    context "('aba', 10)" do
      it "7" do
        expect(RepeatedString.repeat('aba', 10)).to eql(7)
      end
    end
    context "('a', 1000000000000)" do
      it "1000000000000" do
        expect(RepeatedString.repeat('a', 1000000000000)).to eql(1000000000000)
      end
    end
  end
end

File: algorithms/interview-preparation-kit-1/repeated-string/php/response/spec/RepeatedString.spec.php
<?php

use \Challenge\RepeatedString;

describe('Repeated String', function () {

  it('number of letter a in the first 10 repeated string', function () {
    expect(RepeatedString::calc('a', 10))->toBe(10);
  });

  it('number of letter a in the first 10 repeated string', function () {
    expect(RepeatedString::calc('b', 10))->toBe(0);
  });

  it('number of letter a in the first 10 repeated string', function () {
    expect(RepeatedString::calc('aba', 10))->toBe(7);
  });

  it('number of letter a in the first 1000000000000 repeated string', function () {
    expect(RepeatedString::calc('a', 1000000000000))->toBe(1000000000000);
  });

});
